hier_config/base.py

```diff
@@ -381,6 +364,20 @@ def _with_tags(
 
         return new_instance
 
+    @overload
+    def _config_to_get_to(
+        self,
+        target: HConfig,
+        delta: HConfig,
+    ) -> HConfig: ...
+
+    @overload
+    def _config_to_get_to(
+        self,
+        target: HConfigChild,
+        delta: HConfigChild,
+    ) -> HConfigChild: ...
+
```

Comment:
Is this just here to make sure that target type, delta type, and the return type are the same? Would a typevar work here or are the bounding options not enough to accomplish what you are trying to do? (also unsure if this is a 3.9 issue...)

Reply:
I was able to convert 3 of the 4 cases where I used the `overload` decorator to TypeVar.